packages/transformers/assemblyscript/src/ASTransformer.js

```diff
@@ -0,0 +1,144 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+import {promises as fsp} from 'fs';
+import path from 'path';
+import {tmpdir} from 'os';
+import {Transform} from 'stream';
+
+const MARKER = 'asc:';
+const PREFIX_MATCHER = /^asc:(.+)$/;
+
+// This special import contains the `compileStreaming` polyfill.
+const SPECIAL_IMPORT = '__rollup-plugin-assemblyscript_compileStreaming';
+
+const compileStreamingCode = `
+  export async function compileStreaming(respP) {
+              if('compileStreaming' in WebAssembly) {
+                return WebAssembly.compileStreaming(respP);
+              }
+              return respP
+                .then(resp => resp.arrayBuffer())
+                .then(buffer => WebAssembly.compile(buffer));
+            }
+        `;
```

Comment:
Not sure how can I use these in Parcel

Reply:
To share code between transformer outputs, you can put this into a separate file inside the transformer package like here:

https://github.com/parcel-bundler/parcel/blob/e183e32c811f066310fff0610aa856ca73a2fcb6/packages/transformers/js/src/visitors/modules.js#L281-L296 which includes
https://github.com/parcel-bundler/parcel/blob/e183e32c811f066310fff0610aa856ca73a2fcb6/packages/transformers/js/src/esmodule-helpers.js